redaxo/src/core/default.config.yml

```diff
@@ -1,5 +1,7 @@
 setup: true
-debug: false
+debug:
+    enabled: false
+    throw_always_exception: false
```

Comment:
`throw_always_exception` oder `always_throw_exception`, was ist da korrekter?
Oder noch ganz andere Ideen? So ganz glücklich bin ich mit dem Namen noch nicht..

Reply:
Vllt errors_to_exceptions

Reply:
Finde ich missverständlich, denn Error (E_PARSE, E_ERROR etc) werden immer in Exceptions umgewandelt.
Mit dieser Option aber zusätzlich auch Warnings, Notices, Strict, Deprecated, falls sie denn überhaupt im error_reporting() enthalten sind. Aber mir fällt kein guter Oberbegriff für diese vier ein. Höchstens halt einfach `notices_to_exceptions`, und man muss halt wissen, dass damit die vier Typen gemeint sind.

Reply:
Vllt sollten wir hier kein bool flag machen sondern eine einstlelung die bestimmt was alles umgewandelt wird?

Man kann ja ie bitflags verbinden dafür?

Reply:
Ich will vermeiden, dass man hier diese Typen alle auflisten muss:
https://github.com/redaxo/redaxo/blob/a1b93a5f1752a1d26f3550d49063fe9ded0b318e/redaxo/src/core/lib/error_handler.php#L167

Ich finde true/false hier praktischer. Und wenn man denn möchte, kann man übers native error_reporting ja auch trotzdem noch manche Typen ganz unterbinden.
Aber mit dieser Einstellung kann man halt sehr leicht sagen, ob man Exceptions nur für die typischen/echten Errors haben möchte, oder eben auch für Warnings/Notices etc.

Reply:
Error per defualt bleibt ja. Ich würde hier nur notice und warning via bitflag anbieten

Reply:
Strict gibts ja in aktuellem php auch nicht mehr

Reply:
Also `errors_to_exceptions` fände ich dann aber immer noch nicht so gut als Namen, denn die "Errors" werden ja nun mal sowieso in Exceptions umgewandelt.

> Man kann ja ie bitflags verbinden dafür?

Aber wie würdest du es in YAML notieren? Als Array und die Konstanten als Strings?
`[E_NOTICE, E_WARNING]`
Und dann PHP-seitig die Konstanten mit `constant()` einlesen und die Bits verknüpfen?
Oder mit [`!php/const`](https://symfony.com/blog/new-in-symfony-3-2-php-constants-in-yaml-files) arbeiten? Wobei man die dann trotzdem als Array notieren müsste.

Ich würde es aber gerne weiterhin so haben wollen, dass man mit `true` einfach alle aktivieren kann (und mit `false` keine außer die Errors die sowieso in Exception umgewandelt werden).

Reply:
> Also errors_to_exceptions fände ich dann aber immer noch nicht so gut als Namen

Vorschlag: `error_level_all: true`

Reply:
Bezeichnung habe ich erstmal so gelassen, finde ich bisher auch noch am passendsten, glaube ich.
Man kann aber nun optional einzelne Error-Level übergeben, z. B.:

```yaml
debug:
    enabled: true
    throw_always_exception: [E_WARNING, E_NOTICE]
```

Reply:
`true` geht aber auch weiterhin